### PR TITLE
adding devtest lab virtual network

### PR DESCRIPTION
**Pull request by zikalino**
This is basic implementation of DevTest Lab Virtual Network.
In basic implementation default virtual network is created.
In more advanced scenario user should be able to refer external Virtual Network resource, and then override it's subnet, but that's pretty complex scenario so I am leaving it for later.

### CODE REVIEW

**Comment by Fred-sun** [comment]
@yungezz @yuwzho

File: library/azure_rm_devtestlabvirtualnetwork.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_devtestlabvirtualnetwork
+version_added: "2.8"
+short_description: Manage Azure DevTest Lab Virtual Network instance.
+description:
+    - Create, update and delete instance of DevTest Lab Virtual Network.
```
Comment (yungezz) [inline library/azure_rm_devtestlabvirtualnetwork.py]: Azure

File: library/azure_rm_devtestlabvirtualnetwork.py [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+'''
+
+RETURN = '''
+id:
+    description:
+        - The identifier of the resource.
+    returned: always
+    type: str
+    sample: "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourcegroups/testrg/providers/microsoft.devtestlab/
+             mylab/labzimsxxxabcmooo/virtualnetworks/myvn"
```
Comment (yungezz) [inline library/azure_rm_devtestlabvirtualnetwork.py]: mylab/labzimsxxxabcmooo/? could you change it to 'mytestlab/xx' if 'xx' is necessary?

File: library/azure_rm_devtestlabvirtualnetwork.py [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+external_provider_resource_id:
+    description:
+        - The identifier of external virtual network.
+    returned: always
```
Comment (yungezz) [inline library/azure_rm_devtestlabvirtualnetwork.py]: return always?

File: library/azure_rm_devtestlabvirtualnetwork.py [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class Actions:
+    NoAction, Create, Update, Delete = range(4)
+
+
+class AzureRMVirtualNetwork(AzureRMModuleBase):
```
Comment (yungezz) [inline library/azure_rm_devtestlabvirtualnetwork.py]: wrong class name, should be conflict with vnet, AzureRMDevTestLabVirtualNetwork

File: library/azure_rm_devtestlabvirtualnetwork.py [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+            elif kwargs[key] is not None:
+                self.virtual_network[key] = kwargs[key]
+
+        response = None
+
+        self.mgmt_client = self.get_mgmt_svc_client(DevTestLabsClient,
```
Comment (yungezz) [inline library/azure_rm_devtestlabvirtualnetwork.py]: api version

File: library/azure_rm_devtestlabvirtualnetwork.py [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        resource_group = self.get_resource_group(self.resource_group)
```
Comment (yungezz) [inline library/azure_rm_devtestlabvirtualnetwork.py]: set  default location?

File: library/azure_rm_devtestlabvirtualnetwork.py [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+                self.to_do = Actions.Delete
+            elif self.state == 'present':
+                if self.virtual_network.get('description') is not None and self.virtual_network.get('description') != old_response.get('description'):
+                    self.to_do = Actions.Update
+
+        if (self.to_do == Actions.Create) or (self.to_do == Actions.Update):
+            self.log("Need to Create / Update the Virtual Network instance")
+
+            if self.check_mode:
+                self.results['changed'] = True
```
Comment (yungezz) [inline library/azure_rm_devtestlabvirtualnetwork.py]: redundant with line 191

File: library/azure_rm_devtestlabvirtualnetwork.py (hunk 2 above)
Comment (zikalino) [reply library/azure_rm_devtestlabvirtualnetwork.py]: yes, changed to mylab

File: library/azure_rm_devtestlabvirtualnetwork.py (hunk 3 above)
Comment (zikalino) [reply library/azure_rm_devtestlabvirtualnetwork.py]: yes, it's always either created or may be assigned (in the future)

**Pull request merged**